src/foremast/pipeline/construct_pipeline_block.py

```diff
@@ -142,6 +142,12 @@ def construct_pipeline_block(env='',
     hc_grace_period = data['asg'].get('hc_grace_period')
     app_grace_period = data['asg'].get('app_grace_period')
     grace_period = hc_grace_period + app_grace_period
+    
+    ssh_keypair = data['asg'].get('ssh_keypair', None)
```

Comment:
@sijis I confirmed, it was a 100% stale reference, whoever was using it, was definitely not using it anywhere. It basically was a stale key in the dictionary.